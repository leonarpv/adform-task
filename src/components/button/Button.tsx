import React, { ButtonHTMLAttributes, DetailedHTMLProps } from 'react'
import classNames from 'classnames'
import './button.scss'

type Props = DetailedHTMLProps<
    ButtonHTMLAttributes<HTMLButtonElement>,
    HTMLButtonElement
> & {
    secondary?: boolean
}

const Button = ({ secondary, ...rest }: Props) => {
    const className = classNames('button', {
        'button--secondary': secondary,
    })

    return <button className={className} {...rest} />
}

export default Button
