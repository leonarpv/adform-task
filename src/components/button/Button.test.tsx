import { shallow } from 'enzyme'
import React from 'react'
import Button from 'components/button/Button'

describe('Button', () => {
    it('should render properly', () => {
        const wrapper = shallow(<Button />)

        expect(wrapper).toMatchSnapshot()
    })

    it('should render button', () => {
        const wrapper = shallow(<Button />)

        expect(wrapper.find('button').exists()).toBe(true)
        expect(wrapper.find('button.button--secondary').exists()).toBe(false)
    })

    it('should render button in secondary state', () => {
        const wrapper = shallow(<Button secondary={true} />)

        expect(wrapper.find('button.button--secondary').exists()).toBe(true)
    })
})
