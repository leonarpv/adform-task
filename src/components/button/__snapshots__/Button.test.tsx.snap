// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render properly 1`] = `ShallowWrapper {}`;
